app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -137,7 +137,7 @@ class SpotlightFragment : InjectableFragment(), SpotlightNavigationListener, Spo
         }
 
         override fun onEnded() {
-          targetList.removeFirst()
+          if (targetList.isNotEmpty()) targetList.removeFirst()
```

Comment:
Is there a test that produces the crash which we can use this to verify as passing?

Reply:
![image](https://user-images.githubusercontent.com/64526117/204085913-6b5e2969-d915-4770-97df-6fbee7178af6.png)

This tests mimics the behavior when the close spotlight button is tapped twice rapidly in a row. The test fails because the close button is not available, not because it is out of elements to pop. This essentially means that the functionality works as expected, but at the moment, more information is required how the crash reported in the email occurred. It might be possible that the crash was due to some unforeseen circumstance/state that was not caught by the error log such as too many apps being run parallelly on the device. In essence, the spotlight library itself has been designed in a way that this error should not occur. A test case cannot be written with the information we have today.